{"pr_number": 4183, "pr_title": "[RW-5761][risk=no] Workbench long sign-in time sometimes is causing Puppeteer tests to fail randomly", "pr_createdAt": "2020-10-16T19:42:03Z", "pr_url": "https://github.com/all-of-us/workbench/pull/4183", "timeline": [{"oid": "936aa69609ab41ff9dc42f484cadadaec7c5e8a6", "url": "https://github.com/all-of-us/workbench/commit/936aa69609ab41ff9dc42f484cadadaec7c5e8a6", "message": "handle hook_failure event", "committedDate": "2020-10-16T19:40:15Z", "type": "commit"}, {"oid": "f0140dbdbe9d3acbe2b01aae316769298b8f0db4", "url": "https://github.com/all-of-us/workbench/commit/f0140dbdbe9d3acbe2b01aae316769298b8f0db4", "message": "wait app-signed-in 60 seconds", "committedDate": "2020-10-16T20:00:35Z", "type": "commit"}, {"oid": "c72dc2e968cb368855188983e1e84c3bb4e4e024", "url": "https://github.com/all-of-us/workbench/commit/c72dc2e968cb368855188983e1e84c3bb4e4e024", "message": "logs path", "committedDate": "2020-10-16T20:11:36Z", "type": "commit"}, {"oid": "e19465359736c77e5a98a4883865ea43cde11732", "url": "https://github.com/all-of-us/workbench/commit/e19465359736c77e5a98a4883865ea43cde11732", "message": "unchange config.yml", "committedDate": "2020-10-16T20:13:33Z", "type": "commit"}, {"oid": "e00d07fa1902470dcbb2cca433216582b62249d0", "url": "https://github.com/all-of-us/workbench/commit/e00d07fa1902470dcbb2cca433216582b62249d0", "message": "[SKIP CI] unchange jest-circus.setup.ts", "committedDate": "2020-10-16T21:15:46Z", "type": "commit"}, {"oid": "2fe54c2e137d53e118c898711350fa499af93ed8", "url": "https://github.com/all-of-us/workbench/commit/2fe54c2e137d53e118c898711350fa499af93ed8", "message": "fail profile-page.spec.ts", "committedDate": "2020-10-16T21:24:40Z", "type": "commit"}, {"oid": "b5d799f80b23aacdf6d93788905ec28579f9b74e", "url": "https://github.com/all-of-us/workbench/commit/b5d799f80b23aacdf6d93788905ec28579f9b74e", "message": "no retry", "committedDate": "2020-10-16T21:25:01Z", "type": "commit"}, {"oid": "8d3e03befeafba281078f93c5f4512a0a15914a7", "url": "https://github.com/all-of-us/workbench/commit/8d3e03befeafba281078f93c5f4512a0a15914a7", "message": "revert", "committedDate": "2020-10-16T21:38:03Z", "type": "commit"}, {"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDUwNjcyNjUyNA==", "url": "https://github.com/all-of-us/workbench/pull/4183#discussion_r506726524", "bodyText": "When issue first happened in CircleCI, there were no screenshot taken. Without screenshot, it was hard to determine what happened. After some digging, issue was because it occurred inside beforeEach.\nbeforeEach(async () => {\n    await signIn(page);\n  });\n\nAdding case 'hook_failure' solved problem.", "author": "aweng98", "createdAt": "2020-10-16T21:21:02Z", "path": "e2e/puppeteer-custom-environment.ts", "diffHunk": "@@ -16,24 +16,29 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {\n \n   // Take a screenshot right after failure\n   async handleTestEvent(event, state) {", "originalCommit": "e00d07fa1902470dcbb2cca433216582b62249d0", "replyToReviewId": null, "replies": null, "type": "inlineReview"}]}